src/Message/Launch/Builder/LtiResourceLinkLaunchRequestBuilder.php

```diff
@@ -55,10 +55,16 @@ public function buildLtiResourceLinkLaunchRequest(
                 ])
             );
 
+            $launchUrl = $ltiResourceLink->getUrl() ?? $registration->getTool()->getLaunchUrl();
+
+            if (null === $launchUrl) {
+                throw new LtiException('Neither resource link url nor tool default url can be used for launch');
```

Comment:
The message feels a bit ambigious. Maybe something like "Neither resource link url nor tool default url were presented" or similar will be more clear here.

Reply:
agreed, fixed